functions/analytics.js

```diff
@@ -10,39 +10,40 @@ const { formatSecondsAsHHMMSS } = require("./src/utils/time");
 const { checkIsAdmin } = require("./src/utils/permissions");
 const { checkAuth } = require("./src/utils/assert");
 
-const { chunk } = lodash;
+const { chunk, flatten } = lodash;
 
 const functionsConfig = functions.config();
 
 const getUsersWithVisits = async (venueIdsArray) => {
-  const dto = await chunk(venueIdsArray, 10)
-    .map(async (idsArray) => {
-      return await admin
-        .firestore()
-        .collection("users")
-        .where("enteredVenueIds", "array-contains-any", idsArray)
-        .get()
-        .then((usersSnapshot) =>
-          usersSnapshot.docs.map(async (userDoc) => {
-            const user = { ...userDoc.data(), id: userDoc.id };
-
-            // eslint-disable-next-line promise/no-nesting
-            const visits = await userDoc.ref
-              .collection("visits")
-              .get()
-              .then((visitsSnapshot) =>
-                visitsSnapshot.docs.map((visitDoc) => ({
-                  ...visitDoc.data(),
-                  id: visitDoc.id,
-                }))
-              );
-
-            return { user, visits };
-          })
-        );
-    })
-    .map((el) => el.then((res) => res.flat()))
-    .flat();
+  const dto = flatten(
```

Comment:
I think this is what's needed https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap . I don't know why lodash utility is preferred over built-in JS functions though.

That being said, the solution that uses built-ins might need tweaks with Webpack and what polyfills it needs, so, for now, a minor suggestion:

```suggestion
  // @debt see if JS built in .flatMap is good fit for this logic (maybe would require polyfill to be set)
  const dto = flatten(
```